### DISCUSSION TITLE
bug: Input type completion fails when using  yarn workspaces

### DISCUSSION THREAD

**Discussion by neuxregime** [trpc/trpc#2903] [Q&A]
### Provide environment information

  System:
    OS: Linux 5.15 Manjaro Linux
    CPU: (32) x64 AMD Ryzen Threadripper 1950X 16-Core Processor
    Memory: 54.29 GB / 62.65 GB
    Container: Yes
    Shell: 5.9 - /usr/bin/zsh
  Binaries:
    Node: 16.14.2 - ~/.nvm/versions/node/v16.14.2/bin/node
    Yarn: 3.2.3 - ~/.nvm/versions/node/v16.14.2/bin/yarn
    npm: 8.5.0 - ~/.nvm/versions/node/v16.14.2/bin/npm
  Browsers:
    Chromium: 106.0.5249.61
    Firefox: 105.0

### Describe the bug

When using **yarn workspaces**, both `client.query` and the `useQuery` hook fail to provide input type autocompletion in vscode, However path completion works as expected.

Given a single router in `packages/trpc/index.ts`

```Typescript
import * as trpc from "@trpc/server";
import { z } from "zod";

type Context = {
  time: number;
};

const createRouter = () => {
  return trpc.router<Context>();
};

const test = createRouter().query("test", {
  input: z.object({
    param: z.string(),
  }),
  output: z.string(),
  async resolve({ ctx }) {
    return "hello";
  },
});

const appRouter = createRouter().merge("auth.", test);
export type AppRouter = typeof appRouter;
```

The following code in `apps/mobile-web/client.ts` is displayed in vscode with no errors but it should show input related errors since `auth.test` requires an input object:
![image](https://user-images.githubusercontent.com/89808147/193506232-9b9224fe-9e3d-414b-92d1-44fb63328266.png)

### Link to reproduction

https://github.com/neuxregime/trpc-bug

### To reproduce

Clone the above repo. 

Run `yarn`.
Open  `apps/mobile-web/client.ts` in vscode.

### 👨‍👧‍👦 Contributing

- [ ] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

**Comment by KATT** [comment]
We have multiple examples using yarn workspaces, including the tRPC repo itself, this is not a bug but a problem with how it's used. 

Hopefully, someone has time to look at your repo to point out what's wrong.

**Reply by neuxregime** [reply]
> We have multiple examples using yarn workspaces, including the tRPC repo itself, this is not a bug but a problem with how it's used.
> 
> Hopefully, someone has time to look at your repo to point out what's wrong.

I managed to fix the issue. The consuming workspace must import `@trpc/server` which is counter intuitive. It'd be nice to reflect this in the docs

**Reply by KATT** [reply]
Please open a PR to the docs in an appropriate place!
